test/fixtures/rules/std_rule_cases/L059.yml

```diff
@@ -0,0 +1,77 @@
+rule: L059
+
+test_fail_inconsistent_capitalisation_1:
+  # Test that we don't have the "inconsistent" bug
+  fail_str: CREATE TABLE table1 (account_id BiGinT);
+  fix_str: CREATE TABLE table1 (account_id BIGINT);
+  configs:
+    rules:
+      L059:
+        capitalisation_policy: upper
+
+test_fail_inconsistent_capitalisation_2:
+  fail_str: CREATE TABLE table1 (account_id BiGinT);
+  fix_str: CREATE TABLE table1 (account_id bigint);
+  configs:
+    rules:
+      L059:
+        capitalisation_policy: lower
+
+test_fail_inconsistent_capitalisation_3:
+  fail_str: CREATE TABLE table1 (account_id BiGinT);
+  fix_str: CREATE TABLE table1 (account_id Bigint);
+  configs:
+    rules:
+      L059:
+        capitalisation_policy: capitalise
+
+test_fail_capitalisation_policy_lower:
+  fail_str: CREATE TABLE table1 (account_id BIGINT);
+  fix_str: CREATE TABLE table1 (account_id bigint);
+  configs:
+    rules:
+      L059:
+        capitalisation_policy: lower
+
+test_fail_capitalisation_policy_lower_2:
+  fail_str: CREATE TABLE table1 (account_id BIGINT, column_two varchar(255));
+  fix_str: CREATE TABLE table1 (account_id bigint, column_two varchar(255));
+  configs:
+    rules:
+      L059:
+        capitalisation_policy: lower
+
+test_fail_capitalisation_policy_upper:
+  fail_str: CREATE TABLE table1 (account_id bigint);
+  fix_str: CREATE TABLE table1 (account_id BIGINT);
+  configs:
+    rules:
+      L059:
+        capitalisation_policy: upper
+
+test_fail_capitalisation_policy_upper_2:
+  fail_str: CREATE TABLE table1 (account_id BIGINT, column_two varchar(255));
+  fix_str: CREATE TABLE table1 (account_id BIGINT, column_two VARCHAR(255));
+  configs:
+    rules:
+      L059:
+        capitalisation_policy: upper
+
+test_fail_capitalisation_policy_capitalise:
+  # Test for capitalised casing
+  fail_str: CREATE TABLE table1 (account_id BIGINT);
+  fix_str: CREATE TABLE table1 (account_id Bigint);
+  configs:
+    rules:
+      L059:
+        capitalisation_policy: capitalise
+
+test_fail_capitalisation_policy_keywords:
+  # Test cases where data types are keywords, not data_type_identifiers
+  # See: https://github.com/sqlfluff/sqlfluff/pull/2121
+  fail_str: CREATE TABLE table1 (account_id BIGINT, column_two serial);
+  fix_str: CREATE TABLE table1 (account_id BIGINT, column_two SERIAL));
+  configs:
+    rules:
+      L059:
+        capitalisation_policy: upper
```

Comment:
Could you also add this as a test to L010 so we can see if that rule changes the capitalisation atm?

Something like:
```yml
test_pass_capitalisation_policy_data_type_keywords:
  # Test cases where data types are keywords and should therefore not be touched by L010
  # See: https://github.com/sqlfluff/sqlfluff/pull/2121
  fail_str: CREATE TABLE table1 (account_id bigint, column_two serial);
  configs:
    rules:
      L010:
        capitalisation_policy: upper
```

Reply:
I tried this and it fails as well.

Reply:
yeah but you should still add it to the tests because:
a) It's convenient for the reviewers
b) How can we know that we've fixed L010s impact on data_type keywords if you don't add it?
c) How can we be sure that someone in the future will make a change that causes L010 to start messing with the data type keywords again after we've fixed them?

Reply:
@jpy-git Added!

Reply:
@jpy-git shouldn't the L010 test be this? 
```
test_pass_capitalisation_policy_data_type_keywords:
  # Test cases where data types are keywords but are handled by L059 instead of L010
  # See: https://github.com/sqlfluff/sqlfluff/pull/2121
  fail_str: create table table1 (account_id bigint, column_two serial);
  fix_str: CREATE TABLE table1 (account_id bigint, column_two serial);
  configs:
    rules:
      L010:
        capitalisation_policy: upper
```
My thought is that L010 would only be affecting the `create table` statement. Since L059 inherits L010, I'm assuming that the logic in L059 wouldn't be picked up by L010, correct?

Reply:
added an explanation in my review